modules/common/src/main/post-df/magic.scala

```diff
@@ -7,6 +7,22 @@ import notebook.front.widgets.isNumber
 
 trait ExtraSamplerImplicits {
   import SamplerImplicits.Sampler
+  /**
+   * count number of rows in a DataFrame, and do sampling if needed
+   */
+  implicit def SimpleDataFrameSampler = new Sampler[DataFrame] {
+    def apply(df: DataFrame, max: Int): DataFrame = {
+      val count = df.cache().count()
+      if (count > max) {
+        println("DF is larger than maxRows. Will use sampling.")
```

Comment:
btw, if you now how to properly print a message so it'd be shown to the end user,  go ahead and fix this.
> from within the sampler, `prinln` don't display any message, it seems.